### PR TITLE
203 implement downloadable for summary view

### LINKED ISSUE #203: Implement Downloadable for Summary View
### Description

This User Story implements the second functionality for Sprint 4. We need a download button at the end of the table of the Summary View that will convert the HTML table to a CSV file directly and have it downloaded on your PC. Format of the CSV is up to you

Refer to this for documentation: https://chatgpt.com/share/6737c621-bc80-8004-8dfa-9b8fc7b11ef0

### PR DESCRIPTION
**Pull request by cl4414**
See comments on issues #203

### CODE REVIEW

File: frontend/src/pages/course/CourseSummaryPage.tsx [hunk 1]
```diff
@@ -45,7 +45,8 @@ export function CourseSummaryPage() {
               <h1 className="text-2xl font-bold text-white">
                 Summary View of {courseCode}
               </h1>
-              <CourseGroupTable courseid={courseid||""}/>
+              <CourseGroupTable courseid={courseid || ""} />
+              <DownloadCSVButton courseid={courseid || ""} />
```
Comment (LamdaDev) [inline frontend/src/pages/course/CourseSummaryPage.tsx]: Great job Calvin, looks great! Only thing I would like to ask for your opinion design wise is whether we should also hide the "Download CSV" button when the table has not loaded yet, such as the example:

image

Feel free to make this modification if you want but your PR is approved!

**Review by LamdaDev — APPROVED** [review]
LGTM! Only 1 small comment left :D

**Comment by cl4414** [comment]
State issue is fixed!
@LamDaniel1 Should be good now, pls check it again :)

**Comment by LamdaDev** [comment]
Perfect! I'll merge it for you

**Pull request merged**